File: backend/model/src/main/java/eu/viandeendirect/model/OrderStatus.java
package eu.viandeendirect.model;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import jakarta.annotation.Generated;

/**
 * Gets or Sets OrderStatus
 */

@Generated(value = "org.openapitools.codegen.languages.SpringCodegen")
public enum OrderStatus {
  
  BOOKED_WITHOUT_PAYMENT("BOOKED_WITHOUT_PAYMENT"),

  PAYMENT_PENDING("PAYMENT_PENDING"),

  PAYMENT_COMPLETED("PAYMENT_COMPLETED"),
  
  PAYMENT_ABORTED("PAYMENT_ABORTED"),
  
  DELIVERED("DELIVERED");

  private String value;

  OrderStatus(String value) {
    this.value = value;
  }

  @JsonValue
  public String getValue() {
    return value;
  }

  @Override
  public String toString() {
    return String.valueOf(value);
  }

  @JsonCreator
  public static OrderStatus fromValue(String value) {
    for (OrderStatus b : OrderStatus.values()) {
      if (b.value.equals(value)) {
        return b;
      }
    }
    throw new IllegalArgumentException("Unexpected value '" + value + "'");
  }
}

File: frontend/app/src/enum/OrderStatus.ts
export enum OrderStatus {
  BookedWithoutpayment = 'BOOKED_WITHOUT_PAYMENT', 
  PaymentPending = 'PAYMENT_PENDING', 
  PaymentCompleted = 'PAYMENT_COMPLETED', 
  PaymentAborted = 'PAYMENT_ABORTED', 
  Delivered = 'DELIVERED'
}

export class OrderStatusUtils {
  static getOrderStatusLabel(orderStatus: OrderStatus) {
    switch (orderStatus) {
      case OrderStatus.BookedWithoutpayment:
        return 'Réservé sans paiement';
      case OrderStatus.PaymentPending:
        return 'Paiement en cours';
      case OrderStatus.PaymentCompleted:
        return 'Payé';
      case OrderStatus.PaymentAborted:
        return 'Paiement échoué';
        case OrderStatus.Delivered:
          return 'Livré';
        default:
        return '';
    }
  }

}